### PR TITLE
Github actions automated release

### PR DESCRIPTION
**Pull request by fubarhouse**
This PR supersedes/replaces automated releases which have previously happened via Travis CI and also replaces the automatic version injection which also happened via Travis.

It will provide automatic updates to `homebrew-pygmy-go` so that `brew` can update automatically when a tag is pushed, and will even run the same workflow on pushes and tags against `master` and all pull requests - in addition to the existing Travis CI work.

I'm generally against squash merges - but this one calls for it.

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+before:
+  hooks:
+    - go mod download
+brews:
+  - github:
+      owner: fubarhouse
+      name: homebrew-pygmy-go
+    folder: Formula
+    homepage: "https://github.com/fubarhouse/pygmy-go"
+    description: "Amazee.io's local development helper tool"
```
Comment (tobybellwood) [inline .goreleaser.yml]:
```suggestion
    description: "amazee.io's local development helper tool"
```

File: .goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    skip_upload: false
+    test: system "#{bin}/pygmy-go version"
+    install: bin.install "pygmy-go"
+    dependencies:
+      - name: go
```
Comment (tobybellwood) [inline .goreleaser.yml]: Does this download a built binary and extract it - if so, is `go` a dependency, or does it build it locally?

Comment (fubarhouse) [reply .goreleaser.yml]: You can see an example of the results here:
https://github.com/fubarhouse/homebrew-pygmy-go/blob/master/dist/pygmy-go.rb

Comment (tobybellwood) [reply .goreleaser.yml]: yup - are those downloads usable binaries - or do they need building once tapped?

Comment (fubarhouse) [reply .goreleaser.yml]: They contain the compiled binaries, but I will be verifying this.
I also need to make sure those binaries contain the correct version number - but right now I am comfortable they do.

Comment (tobybellwood) [reply .goreleaser.yml]: we can always remove go locally to test if we need to - I think having this set will also download go via brew anyway.

Comment (fubarhouse) [reply .goreleaser.yml]: Agreed, the dependency should not be needed.
I'ill do some more testing tonight when I can actually work on this again.

**Pull request merged**